lib/spack/spack/binary_distribution.py

```diff
@@ -239,25 +239,31 @@ def build_tarball(spec, outdir, force=False, rel=False, yes_to_all=False,
             os.remove(specfile_path)
         else:
             raise NoOverwriteException(str(specfile_path))
-    # make a copy of the install directory to work with
-    workdir = join_path(outdir, os.path.basename(spec.prefix))
-    if os.path.exists(workdir):
-        shutil.rmtree(workdir)
-    install_tree(spec.prefix, workdir)
 
     # create info for later relocation and create tar
-    write_buildinfo_file(workdir)
-
-    # optinally make the paths in the binaries relative to each other
+    write_buildinfo_file(spec.prefix)
```

Comment:
Is there a reason you started writing the build info file directly to the prefix instead of to a copy of the prefix? This results in `binary_distribution` files which exist in the packages installed to the `opt` directory. That's not necessarily a big deal, although it would make it hard to distinguish between packages that have been installed from a binary cache and packages that have been built locally.

Reply:
I decided to only make the copy when I was doing relative paths. The copy was taking a long time. I guess I can remove the binary_distribution file after the tarball is created.